cmd/release-job-migrator/main.go

```diff
@@ -0,0 +1,460 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	prowconfig "k8s.io/test-infra/prow/config"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	configlib "github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/jobconfig"
+)
+
+type GeneratorConfig struct {
+	// Configs defines the configuration needed for a specified test name
+	Configs map[string]testConfig `json:"config"`
+}
+
+type testConfig struct {
+	// Steps defines the multistage test step configuration to use for this job
+	Steps *api.MultiStageTestConfiguration `json:"steps"`
+	// BaseImages defines any images that this test relies on. Omit the "Name" and/or
+	// "Namespace" fields to have them autofilled based on the job name
+	BaseImages map[string]api.ImageStreamTagReference `json:"base_images,omitempty"`
+}
+
+type testsAndBaseImages struct {
+	tests      []api.TestStepConfiguration
+	baseImages map[string]api.ImageStreamTagReference
+}
+
+type jobInfo struct {
+	As      string
+	Product string
+	Version string
+}
+
+type options struct {
+	config        string
+	ciopConfigDir string
+	rcConfigDir   string
+	jobDir        string
+}
+
+func gatherOptions() (options, error) {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.config, "config", "", "Path to config file")
+	fs.StringVar(&o.ciopConfigDir, "ci-op-configs", "", "Path to ci-operator config files")
+	fs.StringVar(&o.rcConfigDir, "rc-configs", "", "Path to release-controller release config files")
+	fs.StringVar(&o.jobDir, "jobs", "", "Path to ci-operator jobs")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return o, fmt.Errorf("failed to parse flags: %w", err)
+	}
+	return o, nil
+}
+
+func validateOptions(o options) error {
+	if len(o.config) == 0 {
+		return errors.New("--config must be set")
+	}
+	if len(o.ciopConfigDir) == 0 {
+		return errors.New("--ci-op-configs must be set")
+	}
+	if len(o.rcConfigDir) == 0 {
+		return errors.New("--rc-configs must be set")
+	}
+	if len(o.jobDir) == 0 {
+		return errors.New("--jobs must be set")
+	}
+	return nil
+}
+
+func getJobInfo(jobName string) (jobInfo, bool) {
+	// release jobs follow the format of "release-openshift-product-installer-testname-version"
+	if !strings.HasPrefix(jobName, "release-openshift-") {
+		//logrus.Infof("Job %s has wrong prefix", jobName)
```

Comment:
commented log lines :)

Reply:
Ah, log lines from when I trying to debug an issue that was actually in a completely different function that I didn't expect. Removed.